soroban-sdk-macros/src/lib.rs

```diff
@@ -58,8 +58,8 @@ fn _symbol_short(crate_path: &str, s: &LitStr) -> TokenStream {
     match Symbol::try_from_small_str(&s.value()) {
         Ok(_) => quote! {{
             #[allow(deprecated)]
-            const symbol: #crate_path::Symbol = #crate_path::Symbol::short(#s);
-            symbol
+            const SYMBOL: #crate_path::Symbol = #crate_path::Symbol::short(#s);
+            SYMBOL
         }}
```

Comment:
Why not this?

```rs
Ok(_) => quote! {
    #crate_path::Symbol::short(#s)
}
```

Reply:
@willemneal and I thought it was probably best to stick with the convention already in the code, but we're curious why the simpler alternative isn't being used.

Reply:
@chadoh can you open a different PR for that? I'll resolve this so we can merge this one for now

Reply:
The `Symbol::short` function call needs to be made in a const context. That's what the macro is creating a short hand for. Assigning the value to a `const` is how we force the expression to be evaluated at compile time. If the expression is executed outside of a const context it'll be evaluated at runtime.

Reply:
Fascinating, thanks @leighmcculloch!